text/0000-unicode-idents.md

```diff
@@ -0,0 +1,162 @@
+> 
+> For some languages, common transliteration systems exist (in particular, for the Latin-based writing systems). For other languages, users have larger difficulties to use Latin to write their native words.
+
+Additionally some math oriented projects may want to use identifiers closely resembling mathematical writing.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Identifiers include variable names, function and trait names and module names. They start with a letter or an underscore and may be followed by more letters, digits and some connecting punctuation.
+
+Examples of valid identifiers are:
+
+* English language words: `color`, `image_width`, `line2`, `Photo`, `_unused`, ...
+* ASCII words in foreign languages: `die_eisenbahn`, `el_tren`, `artikel_1_grundgesetz`
+* words containing accented characters: `garçon`, `hühnervögel`
+* identifiers in other scripts: `Москва`, `東京`, ...
+
+Examples of invalid identifiers are:
+
+* Keywords: `impl`, `fn`, `_` (underscore), ...
+* Identifiers starting with numbers or containing "non letters": `42_the_answer`, `third√of7`, `◆◆◆`, ...
+* Emojis: 🙂, 🦀, 💩, ...
+
+Similar Unicode identifiers are normalized: `a1` and `a₁` (a&lt;subscript 1&gt;) refer to the same variable. This also applies to accented characters which can be represented in different ways.
+
+To disallow any Unicode identifiers in a project (for example to ease collaboration or for security reasons) limiting the accepted identifiers to ASCII add this lint to the `lib.rs` or `main.rs` file of your project:
+
+```rust
+#![forbid(unicode_idents)]
+```
+
+Some Unicode character look confusingly similar to each other or even identical like the Latin **A** and the Cyrillic **А**. The compiler may warn you about easy to confuse names in the same scope. If needed (but not recommended) this warning can be silenced with a `#[allow(confusable_unicode_idents)]` annotation on the enclosing function or module.
+
+## Usage notes
+
+All code written in the Rust Language Organization (*rustc*, tools, std, common crates) will continue to only use ASCII identifiers and the English language.
+
+For open source crates it is recommended to write them in English and use ASCII-only. An exception should be made if the application domain (e.g. math) benefits from Unicode and the target audience (e.g. for a crate interfacing with Russian passports) is comfortable with the used language and characters. Additionally crates should provide an ASCII-only API.
+
+Private projects can use any script and language the developer(s) desire. It is still a good idea (as with any language feature) not to overuse it.
```

Comment:
I would perhaps weaken the language in these two paragraphs, with phrases like “suggested” and “should consider”. English is indeed the de-facto international language and ASCII-only idents are indeed more friendly to an international audience (if only for typing), but it is not a Rust RFC’s place to judge which other concerns are or are not acceptable reasons to do otherwise, or how much use is overuse.